infrastructure/crypto/src/ristretto/ristretto_keys.rs

```diff
@@ -266,6 +266,11 @@ impl PublicKey for RistrettoPublicKey {
         let p = RistrettoPoint::multiscalar_mul(s, p);
         RistrettoPublicKey::new_from_pk(p)
     }
+
+    /// Generate a shared secret from one party's private key and another party's public key
+    fn shared_secret(k: &Self::K, pk: &Self) -> Self {
```

Comment:
Instead of this, just 
`impl DiffieHellmanSharedSecret for RistrettoPublicKey`

Reply:
Applied change